src/main/java/controller/LottoController.java

```diff
@@ -1,30 +1,43 @@
 package controller;
 
-import domain.Lotto;
+import domain.*;
 import util.LottoSeller;
-import domain.Money;
-import domain.LottoResult;
 import view.LottoInputView;
 import view.LottoOutputView;
 
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class LottoController {
 
+    private final String separator = ",";
+
     public void run() {
         final Money money = new Money(LottoInputView.inputPurchaseAmount());
         List<Lotto> lottoList = LottoSeller.buyLotto(money);
 
         LottoOutputView.printLottoCount(lottoList.size());
         LottoOutputView.printLottoList(lottoList);
 
-        Lotto winningLotto = new Lotto(LottoInputView.inputWinningLotto());
+        WinningLotto winningLotto = inputWinningLotto();
 
         List<LottoResult> results = lottoList.stream()
-                .map(lotto -> LottoResult.findBy(winningLotto, lotto))
+                .map(winningLotto::getResult)
                 .collect(Collectors.toList());
 
-        LottoOutputView.printLottoResult(results);
+        LottoOutputView.printLottoResult(new LottoStatistics(results));
+    }
+
+    private WinningLotto inputWinningLotto() {
+        final List<LottoNumber> winningNumbers = Stream.of(LottoInputView.inputWinningLotto().split(separator))
+            .map(String::trim)
+            .map(LottoNumber::new)
```

Comment:
성능에 영향이 없다면 가독성을 위해 map이나 filter에 함수를 한 번에 작성하지 않고 추가로 사용하는 경향이 있기는 한 것 같아요
하지만 로직이 간단하다 보니 map을 한 번 더 사용해서 O(N)이 추가로 발생하는 것은 불필요해 보입니다.

```suggestion
            .map(LottoNumber::new)
```

trim을 LottoNumber 생성자 내부에서 처리하도록 방어로직을 작성하는 방법도 있겠네요

Reply:
제가 체이닝방식으로 여러번 스트림 중간연산을 사용한 부분은 하나의 중간연산 메서드에 여러가지 로직이 들어가면 가독성측면에서 득보다 실이 많다고 생각하기 때문입니다. 메서드참조 형식으로 함수형 로직을 작성하면 한눈에 어떤식으로 데이터를 가공하는지 보기  쉽고, 이는 O(N)을 한번 더 한다고 하더라도 성능의 차이보다는 버그방지와 가독성측면에서 득이 많을거라고 보여요. 

생성자에서 trim을 해주는 방안은 생성자의 역할을 벗어난다고 생각합니다. 적절하게 validation된 값을 이용하여 생성하여야하는데, 임의적으로 생성자에서 `trim()`을 호출한다면 의도하지 않은 동작이 발생할 수도 있을것으로 보입니다. 생성자에 로직을 넣는다면 공백이 있을 시 `NumberFormatExcetpion`을 throw해주는게 적절한 방안이라 생각하는데 리뷰어님 의견은 어떠신가요?

Reply:
적극적인 의견 내주셔서 정말 감사합니다.

생성자 내 가공은 O(N) 방법을 줄일 수 없을까? 하는 단편적인 생각으로부터 나온 답변 같아요
실제로 기본으로 제공하고 있는 Integer 래퍼 클래스만 봐도 잘못된 값이 들어오면 예외를 던져주고 있는 것처럼
정준님 말씀대로 LottoNumber에서도 적절한 예외를 던지고 외부에서 처리하는 방안이 옳은 것 같습니다.